spec/models/data/summative_quiz_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe Data::SummativeQuiz do
+  let(:user_1) { create(:user, :registered, :agency_setting, role_type: 'childminder') }
+  let(:user_2) { create(:user, :registered, :agency_setting, role_type: 'childminder') }
+
+  let(:assessment_pass_1) { create(:user_assessment, :passed, user_id: user_1.id, score: 100, module: 'module_1') }
+  let(:assessment_fail_1) { create(:user_assessment, :failed, user_id: user_1.id, score: 0, module: 'module_1') }
+  let(:assessment_pass_2) { create(:user_assessment, :passed, user_id: user_2.id, score: 80, module: 'module_1') }
+  let(:assessment_fail_2) { create(:user_assessment, :failed, user_id: user_2.id, score: 0, module: 'module_1') }
+
+  let(:formative_assessment) { create(:user_assessment, user_id: user_1.id, assessments_type: 'formative_assessment') }
+
+  describe '.average_pass_scores_csv' do
+    context 'when summative assessments exist with a passed status' do
+      it 'generates csv with average pass score for each module' do
+        expect(user_1.registration_complete).to eq(true)
+        expect(assessment_pass_1.score).to eq('100')
+        expect(described_class.average_pass_scores_csv).to eq("Module,Average Pass Score\nmodule_1,100.0\n")
+      end
+    end
+
+    context 'when summative assessments exist with a failed status' do
+      it 'generates csv with average pass score for each module' do
+        expect(user_1.registration_complete).to eq(true)
+        expect(assessment_pass_1.score).to eq('100')
+        expect(assessment_fail_1.score).to eq('0')
+        expect(described_class.average_pass_scores_csv).to eq("Module,Average Pass Score\nmodule_1,100.0\n")
+      end
+    end
+  end
+
+  describe '.high_fail_questions_csv' do
+    context 'when summative assessments exist' do
+      let(:answer_correct_1) { create(:user_answer, :correct, :summative, module: 'module_1', name: 'q1', questionnaire_id: 0) }
```

Comment:
Questionnaires shouldn't exist (but they still do until we are done migrating). You could write this another way so you don't have to set the value to `0` every time.